### PR TITLE
feat: Add function to StructArray that returns the Arrow schema

### LINKED ISSUE #187: Large schema mismatches in RecordBatch to StructArray conversion are hard to debug
When turning an Arrow `RecordBatch` into a Narrow `StructArray`, if there is a schema mismatch, there is currently a panic, e.g.:
```
PrimitiveArray expected data type Int64 got Int32
```
I believe this message originates from Arrow, not Narrow.

With structs that have many (nested) fields, it is difficult to figure out which field exactly is wrongly typed, since this message only contains the type and not the field name.

It would be great if the panic message would contain the field name.

Perhaps this is not trivial to implement, so alternatively it would be useful to be able to call something like `fn schema() -> arrow_schema::Schema` on the Narrow `StructArray` such that users could perform run-time schema validation before trying to convert. I haven't done a thorough search, but from a quick glance, I don't see anything that would specifically report schema mismatches in a friendlier way in `arrow_schema`, yet I can imagine adding such functionality. This could then be leveraged with the proposed `schema` function (and prevent panics).

### PR DESCRIPTION
**Pull request by johanpel**
This PR adds the `schema()` function to `StructArray` that provides the means to obtain an `arrow_schema::Schema`. This is useful if you want to compare e.g. the schema of some Arrow `RecordBatch` before wrapping it as a Narrow `StructArray`.

This can be implemented in a more elegant way, which would prevent the allocations of buffers, but this works fine for now.

Also closes #187 well enough (for me).

### CODE REVIEW

File: src/arrow/array/struct.rs [hunk 1]
```diff
@@ -156,6 +156,27 @@ where
...
+        Vec<Arc<(dyn arrow_array::Array + 'static)>>:
+            From<<T as StructArrayType>::Array<VecBuffer>>,
+    {
+        let dummy_slice: [T; 0] = [];
+        let dummy_array = dummy_slice.into_iter().collect::<StructArray<T>>();
+        let dummy_batch = arrow_array::RecordBatch::from(dummy_array);
+        dummy_batch.schema()
+    }
+}
+
```
Comment (mbrobbel) [inline src/arrow/array/struct.rs:179]:
```suggestion
impl<T: StructArrayType, const NULLABLE: bool, Buffer: BufferType> StructArray<T, NULLABLE, Buffer>
where
    <T as StructArrayType>::Array<Buffer>: Validity<NULLABLE> + StructArrayTypeFields,
{
    /// Return the Arrow schema using the fields of this `StructArray`.
    #[must_use]
    pub fn schema() -> arrow_schema::Schema {
        arrow_schema::Schema::new(<<T as StructArrayType>::Array<Buffer> as StructArrayTypeFields>::fields())
    }
}
```

File: tests/derive.rs [hunk 2]
```diff
@@ -257,7 +254,7 @@ mod tests {
 
                 #[derive(ArrayType, Default)]
                 struct FooBar {
-                    foo: bool,
+                    fuu: bool,
```
Comment (mbrobbel) [inline tests/derive.rs:257]: Just curious, why was this changed?

**Review by mbrobbel — COMMENTED** [review]
Thanks, this seems useful! I added one suggestion.

File: tests/derive.rs (hunk 2 above)
Comment (johanpel) [reply tests/derive.rs:257]: Clippy started complaining about it in CI:
https://github.com/mbrobbel/narrow/actions/runs/10737811822/job/29780070914

File: src/arrow/array/struct.rs (hunk 1 above)
Comment (johanpel) [reply src/arrow/array/struct.rs]: Ah, much simpler, thanks

File: tests/derive.rs (hunk 2 above)
Comment (mbrobbel) [reply tests/derive.rs:257]: Super pedantic. I ❤️ Clippy.

**Review by mbrobbel — APPROVED** [review]
